content/browser/renderer_host/cross_process_frame_connector.h

```diff
@@ -85,12 +90,80 @@ class RenderWidgetHostViewChildFrame;
 // SiteInstance, A2 in the picture above. When a child frame navigates in a new
 // process, SetView() is called to update to the new view.
 //
+// For SecureEmbed scenarios:
+//
+// Similar to the OOPIF case, a CrossProcessFrameConnector is used to connect
+// the RenderWidgetHostViewChildFrame in the guest WebContents to the
+// SecureEmbed plugin as its embedder. However, unlike the OOPIF case, there is
+// no RenderFrameProxyHost involved; the CrossProcessFrameConnector is owned
+// directly by the GuestFrameImpl. To avoid code duplication, the same
+// CrossProcessFrameConnector class is used in both scenarios, with context-
+// specific behavior (dependency on the RenderFrameProxyHost in the case of
+// OOPIF and dependency on the GuestFrameImpl in the case of SecureEmbed)
+// handled via the Delegate interface.
 class CONTENT_EXPORT CrossProcessFrameConnector
-    : public CrossProcessFrameConnectorBase {
+    : public input::ChildFrameInputHelper::Delegate {
  public:
-  // |frame_proxy_in_parent_renderer| corresponds to A2 in the example above.
-  explicit CrossProcessFrameConnector(
-      RenderFrameProxyHost* frame_proxy_in_parent_renderer);
+  class Delegate {
+   public:
+    virtual ~Delegate() = default;
+
+    // Called when the child frame has updated its visual properties and needs
+    // to notify the embedder.
+    virtual void DidUpdateVisualProperties(
+        const cc::RenderFrameMetadata& metadata) = 0;
+
+    // Disables auto-resize mode for the child frame.
+    virtual void DisableAutoResize() = 0;
+
+    // Enables auto-resize mode for the child frame, allowing it to
+    // automatically resize itself within the given constraints.
+    virtual void EnableAutoResize(const gfx::Size& min_size,
+                                  const gfx::Size& max_size) = 0;
+
+    // Get the child frame host.
+    virtual RenderFrameHostImpl* GetChildRenderFrameHost() const = 0;
+
+    // TODO(secure-embed): Unclear if we should support parent/root view access.
+    // Get parent/root views.
+    virtual RenderWidgetHostViewBase* GetParentView() = 0;
+    virtual RenderWidgetHostViewBase* GetRootView() = 0;
+
+    // Returns the parent process of the child frame.
+    virtual RenderProcessHost* GetParentProcess() const = 0;
+
+    // Returns the visibility state of the embedder (WebContents-level
+    // visibility).
+    virtual Visibility GetEmbedderVisibility() = 0;
+
+    // Notifies the embedder that the child process has terminated.
+    virtual void ChildProcessGone() = 0;
+
+    // Notifies the embedder that the child frame needs to be reloaded.
+    virtual void NeedsReload() = 0;
+
+    // Notifies the embedder when the child's visibility changes.
+    virtual bool VisibilityChanged(
+        RenderWidgetHostViewChildFrame* view,
+        blink::mojom::FrameVisibility visibility) = 0;
+
+    // TODO(secure-embed): Ideally this wouldn't be a method on the delegate
```

Comment:
If not through a delegate method, how can the intrinsic size be sent to the outer frame?